src/sentry/static/sentry/app/components/events/spanStatusBreakdown.tsx

```diff
@@ -0,0 +1,236 @@
+      return event as SentryTransactionEvent;
+    }
+
+    return undefined;
+  }
+
+  generateBreakdown(): SpanStatusBreakdownType {
+    const event = this.getTransactionEvent();
+
+    if (!event) {
+      return [];
+    }
+
+    const traceContext: TraceContextType | undefined = event?.contexts?.trace;
+
+    if (!traceContext) {
+      return [];
+    }
+
+    const spanEntry: SpanEntry | undefined = event.entries.find(
+      (entry: {type: string}) => entry.type === 'spans'
+    );
+
+    const spans: RawSpanType[] = spanEntry?.data ?? [];
+
+    const rootSpanStatus = traceContext.status;
+    const hasRootSpanStatus =
+      typeof rootSpanStatus === 'string' && rootSpanStatus.length > 0;
+
+    // add transaction root span
+    spans.push({
+      op: traceContext.op,
+      timestamp: event.endTimestamp,
+      start_timestamp: event.startTimestamp,
+      trace_id: traceContext.trace_id || '',
+      span_id: traceContext.span_id || '',
+      status: hasRootSpanStatus ? rootSpanStatus : undefined,
+      data: {},
+    });
+
+    // calculate total number of occurrences of span statuses
+
+    const spanStatusTotals = spans.reduce(
+      (totals: SpanStatusTotals, span: RawSpanType) => {
+        let spanStatus: SpanStatusName | undefined = span.status;
+
+        if (!spanStatus) {
+          spanStatus = noStatus;
+        }
+
+        const total = totals[spanStatus] as number | undefined;
+
+        if (typeof total !== 'number') {
+          totals[spanStatus] = 1;
+          return totals;
+        }
+
+        totals[spanStatus] = total + 1;
+        return totals;
+      },
+      {} as SpanStatusTotals
+    );
+
+    const entries = Object.entries(spanStatusTotals);
+
+    // Object.entries() will not return entries whose keys are Symbols.
+    // We add them here if noStatus exists.
+    if (noStatus in spanStatusTotals) {
+      entries.push([t('no status'), spanStatusTotals[noStatus]]);
+    }
+
+    const sortedSpanStatusBreakdown = entries.sort(
+      (first: [string, number], second: [string, number]) => {
+        const firstTotal = first[1];
+        const secondTotal = second[1];
+
+        if (firstTotal === secondTotal) {
+          return 0;
+        }
+
+        if (firstTotal < secondTotal) {
+          // sort second before first
+          return 1;
+        }
+
+        // otherwise, sort first before second
+        return -1;
+      }
+    );
+
+    // the total number of span statuses (including no status) is equvalent to the
+    // number of spans
+    const totalSum = spans.length;
+
+    return sortedSpanStatusBreakdown.slice(0, TOP_N_SPAN_STATUS).map(
+      ([spanStatus, total]: [string, number]): SpanStatusStats => {
+        return {
+          name: spanStatus,
+          percentage: total / totalSum,
+          total,
+        };
+      }
+    );
+  }
+
+  render() {
+    const event = this.getTransactionEvent();
+
+    if (!event) {
+      return null;
+    }
```

Comment:
Do you need to fetch the event here? The `generateBreakdown()` method does a similar set of checks and returns early when it is missing data.